Mathlib/Mathport/Syntax.lean

```diff
@@ -0,0 +1,727 @@
+syntax (name := simpa!?) "simpa!?" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" Tactic.simpArg,* "]")? (" with " (colGt ident)+)? (" using " term)? : tactic
+
+syntax (name := splitIfs) "splitIfs" (ppSpace location)? (" with " binderIdent+)? : tactic
+
+syntax (name := squeezeScope) "squeezeScope " tacticSeq : tactic
+syntax (name := squeezeSimp) "squeezeSimp" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeSimp?) "squeezeSimp?" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeSimp!) "squeezeSimp!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeSimp?!) "squeezeSimp?!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeSimpa) "squeezeSimpa" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " term)? : tactic
+syntax (name := squeezeSimpa?) "squeezeSimpa?" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " term)? : tactic
+syntax (name := squeezeSimpa!) "squeezeSimpa!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " term)? : tactic
+syntax (name := squeezeSimpa?!) "squeezeSimpa?!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " term)? : tactic
+syntax (name := squeezeDSimp) "squeezeDSimp" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeDSimp?) "squeezeDSimp?" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeDSimp!) "squeezeDSimp!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+syntax (name := squeezeDSimp?!) "squeezeDSimp?!" (" (" &"config" " := " term ")")? (&" only")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (ppSpace location)? : tactic
+
+syntax (name := suggest) "suggest" (" (" &"config" " := " term ")")? (ppSpace num)?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " (colGt ident)+)? : tactic
+-- Moved to Mathlib.Tactic.LibrarySearch
+-- syntax (name := librarySearch) "librarySearch" (" (" &"config" " := " term ")")?
+--   (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " (colGt ident)+)? : tactic
+syntax (name := librarySearch!) "librarySearch!" (" (" &"config" " := " term ")")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " (colGt ident)+)? : tactic
```

Comment:
```suggestion
```

These transient tactics are not (supposed to be!) used in mathlib and they will almost certainly look different in Lean 4, I really don't think we should spend time on porting their syntax 1-to-1.

We could also append `3` to all names, then the syntax doesn't matter and we can just remove the dangling syntax after the port.

But this is not a blocker, none of the syntaxes clash now.

Reply:
I would prefer that the syntaxes continue to exist, although as you say they can be renamed if you want them to look different. But why wouldn't we want them to have a grammar like this? If someone is using `librarySearch!` they shuold be able to add all these config options and so on, I don't see a compelling reason not to have the tactic and a roughly equivalent syntax, not for porting but for lean 3 parity.

Reply:
As discussed in the other comment, the functionality should certainly be available in Lean 4 but there's no point in spending any effort on porting them since they don't occur in mathlib.

Since you've mentioned the bang-version of librarySearch.  The bang-less version (doing only reducible matching) is an optimization we had to do in Lean 3 because semireducible matching was too slow.  We can hopefully eliminate this variant entirely in Lean 4 using better indexing.